src/NHibernate/Dialect/Schema/PostgreSQLMetadata.cs

```diff
@@ -138,7 +138,43 @@ public PostgreSQLColumnMetadata(DataRow rs)
 			this.SetNumericalPrecision(rs["NUMERIC_PRECISION"]);
 
 			Nullable = Convert.ToString(rs["IS_NULLABLE"]);
-			TypeName = Convert.ToString(rs["DATA_TYPE"]);
+			TypeName = NormalizeTypeNames(Convert.ToString(rs["DATA_TYPE"]));
+		}
+
+		private static string NormalizeTypeNames(string typeName)
+		{
+			switch (typeName)
+			{
+				case "double precision":
+					return "float8";
+				case "real":
+					return "float4";
+				case "smallint":
+					return "int2";
+				case "integer":
+					return "int4";
+				case "bigint":
+					return "int8";
+			}
```

Comment:
We can see the naming changes in [Npgsql change commit](https://github.com/npgsql/npgsql/pull/1945/files?diff=split&w=1), or infer it from [PostgreSQL documentation](https://www.postgresql.org/docs/current/datatype.html#DATATYPE-TABLE).

It looks like we were already having an issue with older Npgsql drivers, because we were using the complete name, `boolean`, while older drivers were previously supposed to yield the short name, `bool`. That trouble would be fixed since we do not translate it to its short name here.

But then we keep an inconsistency: the dialect and now the metadata are using PostgreSQL type short aliases for most types excepted for the boolean type.

For the same reasons than in this [comment](https://github.com/nhibernate/nhibernate-core/pull/2992#issuecomment-1013960183), normalizing this would be breaking. So, let it stay that way.